src/v2/v2_types.rs

```diff
@@ -0,0 +1,185 @@
+use chrono::TimeZone;
+use thiserror::Error;
+use crate::endpoints::QueryError;
+
+#[derive(Debug)]
+/// TODO: Someone remind me to doc this type.
+pub struct NetworkStatus {
+    pub node_id:             String,
+    pub peer_total_sent:     u64,
+    pub peer_total_received: u64,
+    pub avg_bps_in:          u64,
+    pub avg_bps_out:         u64,
+}
+
+#[derive(Debug)]
+pub enum NodeType {
+    Bootstrapper, 
+    Node,
+}
+
+#[derive(Debug)]
+/// TODO: Someone remind me to doc this type.
+pub struct NodeInfo {
```

Comment:
Is it `Info` or `Status`? You use both.

Reply:
Please lets stick with `Info`, that is at least what I have done so far

Reply:
The proposal mentions 'Status'. However I could flatten the structure a bit and get rid of the current "Info'. 
I don't care whether it's 'status' or 'info', just that it would make sense to have it somewhat aligned with the name of the grpc call.

Reply:
Then make the call `GetNodeInfo`. I think that is the proposal discussed here. 

As I mentioned the feature proposal is not be all end all. The details are not specified there.

Reply:
Will do